control-plane-operator/controllers/hostedcontrolplane/cno/clusternetworkoperator.go

```diff
@@ -0,0 +1,251 @@
+	KubeRBACProxy                string
+	Multus                       string
+	MultusAdmissionController    string
+	CNIPlugins                   string
+	BondCNIPlugin                string
+	WhereaboutsCNI               string
+	RouteOverrideCNI             string
+	MultusNetworkPolicy          string
+	OVN                          string
+	EgressRouterCNI              string
+	KuryrDaemon                  string
+	KuryrController              string
+	NetworkMetricsDaemon         string
+	NetworkCheckSource           string
+	NetworkCheckTarget           string
+	CloudNetworkConfigController string
+	TokenMinter                  string
+}
+
+type Params struct {
+	ReleaseVersion          string
+	AvailabilityProberImage string
+	HostedClusterName       string
+	APIServerAddress        string
+	APIServerPort           int32
+	TokenAudience           string
+	Images                  Images
+	OwnerRef                config.OwnerRef
+	DeploymentConfig        config.DeploymentConfig
+}
+
+func NewParams(hcp *hyperv1.HostedControlPlane, version string, images map[string]string, setDefaultSecurityContext bool) Params {
+
+	p := Params{
+		Images: Images{
+			NetworkOperator:              images["cluster-network-operator"],
+			SDN:                          images["sdn"],
+			KubeProxy:                    images["kube-proxy"],
+			KubeRBACProxy:                images["kube-rbac-proxy"],
+			Multus:                       images["multus-cni"],
+			MultusAdmissionController:    images["multus-admission-controller"],
+			CNIPlugins:                   images["container-networking-plugins"],
+			BondCNIPlugin:                images["network-interface-bond-cni"],
+			WhereaboutsCNI:               images["multus-whereabouts-ipam-cni"],
+			RouteOverrideCNI:             images["multus-route-override-cni"],
+			MultusNetworkPolicy:          images["multus-networkpolicy"],
+			OVN:                          images["ovn-kubernetes"],
+			EgressRouterCNI:              images["egress-router-cni"],
+			KuryrDaemon:                  images["kuryr-cni"],
+			KuryrController:              images["kuryr-controller"],
+			NetworkMetricsDaemon:         images["network-metrics-daemon"],
+			NetworkCheckSource:           images["cluster-network-operator"],
+			NetworkCheckTarget:           images["cluster-network-operator"],
+			CloudNetworkConfigController: images["cloud-network-config-controller"],
+			TokenMinter:                  images["token-minter"],
+		},
+		ReleaseVersion:          version,
+		AvailabilityProberImage: images[util.AvailabilityProberImageName],
+		OwnerRef:                config.OwnerRefFrom(hcp),
+	}
+
+	p.DeploymentConfig.Scheduling.PriorityClass = config.DefaultPriorityClass
+	p.DeploymentConfig.SetColocation(hcp)
+	p.DeploymentConfig.SetRestartAnnotation(hcp.ObjectMeta)
+	p.DeploymentConfig.SetReleaseImageAnnotation(hcp.Spec.ReleaseImage)
+	p.DeploymentConfig.SetControlPlaneIsolation(hcp)
+	p.DeploymentConfig.Replicas = 1
```

Comment:
should this honour ControllerAvailabilityPolicy?

Reply:
CNO isn't "critical", and any downtime while it gets rescheduled would be acceptable. We only run one in "standard" clusters - is there any reason to be different? (n.b. ingress operator also runs one)